# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxHarfBuzzShaper.cpp
# Commit: a4ad6f718272
# Full Hash: a4ad6f7182724232bd9ba68bc85fb3167beb1335
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2025-05-07 04:19:47
# Regressor Bug: 1961387
# File Overlap Count: 1
# Description:
#   Bug 1961387 - Don't return a vertical-origin of zero for 'empty' truetype glyphs; fall back to computing the default vertical origin, so that if there's color-glyph data it will render in the right place. r=gfx-reviewers,lsalzman
#   
#   This fixes the bad positioning of Noto Color Emoji glyphs when using the
#   current (COLRv1) version of the font. GetGlyphVOrigin was incorrectly
#   returning zero because the 'glyf' data was empty; but the glyph will in
# ==============================================================================

diff -r 15a6b8e475a0 -r a4ad6f718272 gfx/thebes/gfxHarfBuzzShaper.cpp
--- a/gfx/thebes/gfxHarfBuzzShaper.cpp	Tue May 06 20:09:16 2025 +0000
+++ b/gfx/thebes/gfxHarfBuzzShaper.cpp	Tue May 06 20:12:45 2025 +0000
@@ -571,12 +571,10 @@
   if (mVmtxTable) {
     bool emptyGlyf;
     const Glyf* glyf = FindGlyf(aGlyph, &emptyGlyf);
-    if (glyf) {
-      if (emptyGlyf) {
-        *aY = 0;
-        return;
-      }
-
+    // If we didn't find any 'glyf' data, fall through to the default below;
+    // note that the glyph might still actually render (via SVG or COLR data),
+    // so we need to provide a reasonable origin.
+    if (glyf && !emptyGlyf) {
       const ::GlyphMetrics* metrics = reinterpret_cast<const ::GlyphMetrics*>(
           hb_blob_get_data(mVmtxTable, nullptr));
       int16_t lsb;
